src/drivers/intel/ssp/mn.c

```diff
@@ -122,9 +122,6 @@ static inline int setup_initial_mclk_source(uint32_t mclk_id,
 
 	/* searching the smallest possible mclk source */
 	for (i = MAX_SSP_FREQ_INDEX; i >= 0; i--) {
-		if (mclk_rate > ssp_freq[i].freq)
```

Comment:
Btw, should we be using ARRAY_SIZE here ?

Reply:
don't think so, the MAX_SSP_FREQ_INDEX is defined for this and used everywhere else.

Reply:
ARRAY_SIZE can only be used if the compiler can figure out the values. Since we use an extern the compiler can't use ARRAY_SIZE here.

sof/drivers/ssp.h:extern const struct freq_table *ssp_freq;